data/src/main/java/com/linkedin/data/schema/grammar/PdlSchemaParser.java

```diff
@@ -756,6 +760,26 @@ private RecordDataSchema parseRecord(
     return properties;
   }
 
+  /**
+   * Sets properties on the provided {@link ComplexDataSchema} using data parsed from the provided
+   * {@link AnonymousTypeDeclarationContext}.
+   *
+   * @param source source to read properties from
+   * @param target target on which to set properties
+   */
+  private void setProperties(AnonymousTypeDeclarationContext source, ComplexDataSchema target)
```

Comment:
can `setDocAndProperties` reuse this?

Reply:
No, since `AnonymousTypeDeclarationContext` and `NamedTypeDeclarationContext` are separate classes with no inheritance relationship. The only other option would be to pass in the parent `TypeDeclarationContext` object, but that would still require that the anonymous and named types have their properties extracted separately.